src/crepe/system/InputSystem.cpp

```diff
@@ -127,57 +129,63 @@ void InputSystem::handle_move(const SDLContext::EventData & event_data,
 
 		bool was_hovering = button.hover;
 		if (button.active
-			&& is_mouse_inside_button(world_mouse_x, world_mouse_y, button, transform)) {
+			&& this->is_mouse_inside_button(world_mouse_x, world_mouse_y, button, transform)) {
 			button.hover = true;
-			if (!was_hovering && button.on_enter) {
-				button.on_enter();
+			if (!was_hovering && button.on_mouse_enter) {
+				button.on_mouse_enter();
 			}
 		} else {
 			button.hover = false;
 			// Trigger the on_exit callback if the hover state just changed to false
-			if (was_hovering && button.on_exit) {
-				button.on_exit();
+			if (was_hovering && button.on_mouse_exit) {
+				button.on_mouse_exit();
 			}
 		}
 	}
 }
 
-void InputSystem::handle_click(const MouseButton & mouse_button, const int & world_mouse_x,
-							   const int & world_mouse_y) {
+void InputSystem::handle_click(const MouseButton & mouse_button, const int world_mouse_x,
+							   const int world_mouse_y) {
 	ComponentManager & mgr = this->component_manager;
 
 	RefVector<Button> buttons = mgr.get_components_by_type<Button>();
 
 	for (Button & button : buttons) {
 		RefVector<Transform> transform_vec
 			= mgr.get_components_by_id<Transform>(button.game_object_id);
-		Transform& transform(transform_vec.front().get());
+		Transform & transform(transform_vec.front().get());
```

Comment:
Ik heb liever dat je dit zo schrijft zodat het duidelijk is dat je geen constructor aanroept
```cpp
Transform & transform = transform_vec.front();
```